apis/iplocation.py

```diff
@@ -0,0 +1,78 @@
+from typing import Union
+from dataclasses import dataclass
+import ipaddress
+import logging
+import requests
+import socket
+
+from base_db_api import BaseDBAPI
+
+_LOGGER = logging.getLogger('app.apis.iplocation')
+
+@dataclass
+class GeolocationResponse:
+    status: str
+    country: str
+    countryCode: str
+    region: str
+    regionName: str
+    city: str
+    zip: str
+    lat: int
+    lon: int
+    timezone: str
+    isp: str
+    org: str
+    asn: str
+    query: str
+
+
+LOCATION_API = 'http://ip-api.com/json/{}'
+
+
+class LocationApiFailedToFetchIpError(Exception):
+    pass
+
+
+class FailedToResolveDomain(Exception):
+    pass
+
+
+class IPLocationDBAPI(BaseDBAPI):
+    UNSAFE_COUNTRY_CODES = ['ir', 'sy', 'lb']
+    def get_location_info(self, ip: Union[ipaddress.IPv4Address, ipaddress.IPv6Address]) -> GeolocationResponse:
```

Comment:
So it seems that the iplocation service can also get some urls.
so I think we can add str to the union.